apps/user/models/user_profile.py

```diff
@@ -140,11 +123,49 @@ def realname(self) -> str:
 
     @cached_property
     def is_official(self) -> bool:
-        return (
-            self.group in UserProfile.OFFICIAL_GROUPS
-            or self.user.email == "new-ara@sparcs.org"
-        )
+        if self.user.email == "new-ara@sparcs.org":
+            return True
+        for group in self.groups():
+            if group.is_official:
+                return True
+        return False
 
     @cached_property
     def is_school_admin(self) -> bool:
-        return self.group == UserProfile.UserGroup.COMMUNICATION_BOARD_ADMIN
+        return self.has_group_by_id(7)  # 7 = Communication board admin
+
+    def groups(self) -> list[Group]:
+        return UserGroup.search_by_user(self.user)
+
+    def has_group(self, group: Group) -> bool:
+        return group in self.groups()
+
+    def has_group_by_name(self, name: str) -> bool:
+        return any([group.name == name for group in self.groups()])
```

Comment:
```suggestion
        return any(group.name == name for group in self.groups())
```
memory에 list 안띄우고 바로 쓸 수 있는 것 같은데 어떤가요?

Reply:
수정하겠습니다.